### PR TITLE
observe proxy: don't change target in noconfig -> noconfig case

### PR DESCRIPTION
**Pull request by stlaz**
causes trouble in https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/530/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws/2467/artifacts/e2e-aws/pods/openshift-kube-apiserver-operator_kube-apiserver-operator-564769bd4f-p8s98_kube-apiserver-operator.log

### CODE REVIEW

**Comment by stlaz** [comment]
The ill behavior can also be observed in `openshift-apiserver-operator` logs, interestingly it does not report Progressing/Degraded

File: pkg/operator/configobserver/proxy/observe_proxy.go [hunk 1]
```diff
@@ -60,6 +60,9 @@ func (f *observeProxyFlags) ObserveProxyConfig(genericListers configobserver.Lis
 		if err := unstructured.SetNestedStringMap(observedConfig, newProxyMap, f.configPath...); err != nil {
 			errs = append(errs, err)
 		}
+	} else if len(currentProxyMap) == 0 {
+		// both are unset
+		return existingConfig, errs
```
Comment (sttts) [inline pkg/operator/configobserver/proxy/observe_proxy.go]: why is this necessary? Because empty map vs. nil is not deep equal?

Comment (stlaz) [reply pkg/operator/configobserver/proxy/observe_proxy.go]: exactly
